lib/std/crypto/salsa20.zig

```diff
@@ -9,11 +9,185 @@ const crypto = std.crypto;
 const debug = std.debug;
 const math = std.math;
 const mem = std.mem;
+const Vector = std.meta.Vector;
 
 const Poly1305 = crypto.onetimeauth.Poly1305;
 const Blake2b = crypto.hash.blake2.Blake2b;
 const X25519 = crypto.dh.X25519;
 
+const Salsa20VecImpl = struct {
+    const Lane = Vector(4, u32);
+    const Half = Vector(2, u32);
+    const BlockVec = [4]Lane;
+
+    fn initContext(key: [8]u32, d: [4]u32) BlockVec {
+        const c = "expand 32-byte k";
+        const constant_le = comptime [4]u32{
+            mem.readIntLittle(u32, c[0..4]),
+            mem.readIntLittle(u32, c[4..8]),
+            mem.readIntLittle(u32, c[8..12]),
+            mem.readIntLittle(u32, c[12..16]),
+        };
+        return BlockVec{
+            Lane{ key[0], key[1], key[2], key[3] },
+            Lane{ key[4], key[5], key[6], key[7] },
+            Lane{ constant_le[0], constant_le[1], constant_le[2], constant_le[3] },
+            Lane{ d[0], d[1], d[2], d[3] },
+        };
+    }
+
+    inline fn rot(x: Lane, comptime n: u5) Lane {
+        return (x << @splat(4, @as(u5, n))) | (x >> @splat(4, @as(u5, 1 +% ~n)));
+    }
+
+    inline fn salsa20Core(x: *BlockVec, input: BlockVec, comptime feedback: bool) void {
```

Comment:
I feel like this `inline` is going to bloat out small builds.
Does it really not inline itself?

Reply:
An explicit `inline` makes a slight difference (`ReleaseFast`):

no inline:
```text
 xsalsa20Poly1305:        554 MiB/s
```

inline:
```text
 xsalsa20Poly1305:        569 MiB/s
```

With `ReleaseSmall`, the size doesn't change at all with and without.